webhooks/ssp_webhook.go

```diff
@@ -32,27 +31,35 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
-	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
 	sspv1beta2 "kubevirt.io/ssp-operator/api/v1beta2"
+	sspv1beta3 "kubevirt.io/ssp-operator/api/v1beta3"
+	"kubevirt.io/ssp-operator/webhooks/convert"
 )
 
+// +kubebuilder:webhook:verbs=create;update,path=/validate-ssp-kubevirt-io-v1beta2-ssp,mutating=false,failurePolicy=fail,groups=ssp.kubevirt.io,resources=ssps,versions=v1beta2,name=validation.v1beta2.ssp.kubevirt.io,admissionReviewVersions=v1,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update,path=/validate-ssp-kubevirt-io-v1beta3-ssp,mutating=false,failurePolicy=fail,groups=ssp.kubevirt.io,resources=ssps,versions=v1beta3,name=validation.v1beta3.ssp.kubevirt.io,admissionReviewVersions=v1,sideEffects=None
+
 var ssplog = logf.Log.WithName("ssp-resource")
 
 func Setup(mgr ctrl.Manager) error {
-	// This is a hack. Using Unstructured allows the webhook to correctly decode different versions of objects.
-	// Controller-runtime currently does not support a single webhook for multiple versions.
-
-	obj := &unstructured.Unstructured{}
-	obj.SetAPIVersion(sspv1beta2.GroupVersion.String())
-	obj.SetKind("SSP")
-
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(obj).
+	err := ctrl.NewWebhookManagedBy(mgr).
+		For(&sspv1beta2.SSP{}).
 		WithValidator(newSspValidator(mgr.GetClient())).
 		Complete()
-}
+	if err != nil {
+		return fmt.Errorf("failed to create webhook for v1beta2.SSP")
+	}
+
+	err = ctrl.NewWebhookManagedBy(mgr).
+		For(&sspv1beta3.SSP{}).
+		WithValidator(newSspValidatorV1Beta3(mgr.GetClient())).
```

Comment:
We have `newSspValidator` and `newSspValidatorV1Beta3`. Can't we just keep using a single `newSspValidator`?

Reply:
Good idea, I've changed the code.